module/Motion/Pid.cpp

```diff
@@ -0,0 +1,46 @@
+/**
+ * @file Pid.cpp
+ * @brief PIDを計算するクラス
+ * @author kawano
+ */
+
+#include "Pid.h"
+
+PidGain::PidGain(double _kp, double _ki, double _kd) : kp(_kp), ki(_ki), kd(_kd) {}
+
+Pid::Pid(double _kp, double _ki, double _kd, double _targetValue)
+    : gain(_kp, _ki, _kd), preDeviation(0.0), integral(0.0), targetValue(_targetValue)
+{
+}
+
+void Pid::setPidGain(double _kp, double _ki, double _kd)
+{
+  gain.kp = _kp;
+  gain.ki = _ki;
+  gain.kd = _kd;
+}
+
+double Pid::calculatePid(double presentValue, double delta)
+{
+  // 0除算を避けるために0の場合はデフォルト周期0.01とする
+  if (delta == 0)
+    delta = 0.01;
+  //現在の偏差を求める
+  double presentDeviation = targetValue - presentValue;
+  //積分の処理を行う
+  integral += presentDeviation * delta;
+  //微分の処理を行う
+  double difference = (presentDeviation - preDeviation) / delta;
+  //前回の偏差を更新する
+  preDeviation = presentDeviation;

```

Comment:
少し、気になった点として、`pre`と略してしまうと`present`と`previous`どちらかわかりづらいので、修正したほうが良いと思います。

```suggestion
  previousDeviation = presentDeviation;
```
or
```suggestion
  prevDeviation = presentDeviation;
```

Reply:
そもそもpresentを使わない方がいい気がする
previousを使うとなると特に紛らわしいから，nowとかcurrentにするべきか？

Reply:
名前を修正する場合は，テストコードの変数名も変えるのを忘れずに

Reply:
今更申し訳ありませんが、nowよりcurrentの方がよさそうです。
nowだと現在時刻とかの意味になる気がします。
[参考](https://hexadrive.jp/hexablog/diary/23671/)